src/main/java/org/spongepowered/api/command/spec/CommandSpec.java

```diff
@@ -254,7 +291,7 @@ public CommandSpec build() {
             }
 
             return new CommandSpec(this.args, this.executor, this.description, this.extendedDescription, this.permission,
```

Comment:
Should there not be a check here that ensures that the `CommandExecutor<T>` and the specified source type match up? It's probably not so much of a problem if you have a `Player` source type and a `CommandExecutor<CommandSource>` executor, but if you have a `Player` source type and a `CommandExecutor<ConsoleSource>` executor, this is still going to be accepted, and the command just won't work at runtime. Worse, you might have not specified the `sourceType` and the executor is of type `CommandExecutor<Player>`, so running as the console will cause a runtime error.

I realise that type erasure makes this difficult though.